cmd/utils.go

```diff
@@ -16,22 +16,48 @@ func max(x, y int) int {
 }
 
 // find all possible chains starting from currentDep
-func getChains(currentDep string, graph map[string][]string, longestPath []string, chains map[int][]string) {
+func getChains(currentDep string, graph map[string][]string, longestPath []string, chains map[int][]string, cycleChains map[int][]string, iter *int) {
 	longestPath = append(longestPath, currentDep)
 	_, ok := graph[currentDep]
 	if ok {
 		for _, dep := range graph[currentDep] {
 			if !contains(longestPath, dep) {
 				cpy := make([]string, len(longestPath))
 				copy(cpy, longestPath)
-				getChains(dep, graph, cpy, chains)
+				getChains(dep, graph, cpy, chains, cycleChains, iter)
+			} else {
+				chains[len(longestPath)] = longestPath
+				cycleChains[*iter] = append(longestPath, dep)
```

Comment:
why is this an int pointer? 🤔

Reply:
I wanted to store each chain in a map since maps act like they're pass by reference in Go (which means they will act like a global variable across recursive function calls).

And for each chain to have a unique key in the map I thought I'll use an int pointer and increment it each time I store something in the map. Please let me know if there is a better way of doing this 😄

Reply:
well if you need pass by reference, why not pointer of slices?

Reply:
Yeah I don't why I forgot about that 😅 Have done that now please have a look.